lib/puppet/feature/base.rb

```diff
@@ -77,6 +77,7 @@ def self.is_implemented
           true
         rescue LoadError => err
           Puppet.debug("CreateSymbolicLink is not available")
+          Puppet.deprecation_warning("Windows 2003 and below versions are deprecated, and will be officially unsupported in the next major Puppet release.  Please plan accordingly and see http://links.puppetlabs.com/deprecate-windows-2003 for more information.")
```

Comment:
We could break this out into it's own check, but we already know that `CreateSymbolicLinkW` is appropriate feature detection for Windows 2003.